5-Heroku/server/models/todoDAO.js

```diff
@@ -0,0 +1,77 @@
+const mongoose = require('mongoose')
+const todoScheme = new mongoose.Schema({
+    todoText: {
+        type: String,
+        required: true
+    },
+    todoId: {
+        type: String,
+        required: true
+    },
+    checked: {
+        type: Boolean,
+        required: true
+    },
+    userId: {
+        type: String,
+        required: true
+    }
+})
+
+const Todo = mongoose.model('Todo', todoScheme)
+
+const modelGetTodoById = async (userId, todoId) => {
+    try {
+        const todo = await Todo.where({userId: userId}).findOne({'todoId': todoId}).lean(true);
+        const resTodo = {
+            todoId: todo.todoId,
+            todoText: todo.todoText,
+            checked: todo.checked
+        }
+        return resTodo;
+    } catch (err) {
+        return { error: "Todo is not found!" }
+    }
+}
+
+const modelGetAllTodos = async (userId) => {
+    try{
+        const todos = await Todo.where({userId: userId}).find()
+        return todos;
+    } catch (err) {
+        console.log(err);
```

Comment:
why swallow the exception? it should propgate back to whom ever called this

Reply:
got it. ill throw (err)